### ISSUE TITLE
Report standard deviation in benchmarks

### ISSUE THREAD

**Issue by Akirathan** [enso-org/enso#7927]
Our [benchmark charts](https://enso-org.github.io/engine-benchmark-results/stdlib-benchs.html) currently displays only *score* which is an average of one benchmark iteration in milliseconds.

Since JMH can, and does, output stddev, it would be nice to:
```[tasklist]
### Tasks
- [ ] Make sure stddev is saved in the XML generated by the JMH benchmark runner
- [ ] Make sure that this stddev is displayed in the charts, if present in the XML artifact.
``` 

stddev is a very important metric as it tells us basically how much we can trust the results for a particular benchmark.

**Comment by radeusgd** [comment]
![image](https://github.com/enso-org/enso/assets/1436948/363e00ad-fd54-4e5d-b1a9-ab9b9b774ea0)

As noted, the std-dev allows us to easier differentiate benchmarks like the navy vs orange ones from this analysis (one has very high variance and the other is very 'stable').

**Comment by radeusgd** [comment]
Just a note that in #8091 [I added displaying stdev within the Enso runner for the benchmarks](https://github.com/enso-org/enso/pull/8091/commits/f832a0ab12fee435c8fdae6d3b9ef5ac5dbf9f7c). This is a bit perpendicular, because this is _not_ the runner used on CI so the ticket remains open - but it is slightly related so I thought it may be worth noting.

**Comment by radeusgd** [comment]
![image](https://github.com/enso-org/enso/assets/1436948/cec3ca74-a66c-4379-aaa8-df8942f10b08)
We can see how the stdev allows us to better judge the benchmarks - for example the high stdev in the Enso variants may show that the warmup time is insufficient and should be made larger _if_ we want to see the peak performance.

**Comment by JaroslavTulach** [comment]
There already is _"List of suspicious benchmarks that violate six sigma"_ at https://enso-org.github.io/engine-benchmark-results/ - isn't this already fixed?

<blockquote><div><strong><a href="https://enso-org.github.io/engine-benchmark-results/">Enso benchmark results</a></strong></div></blockquote>

**Comment by Akirathan** [comment]
> There already is _"List of suspicious benchmarks that violate six sigma"_ at https://enso-org.github.io/engine-benchmark-results/ - isn't this already fixed?

Not really. The "List of suspicious benchmarks that violate six sigma" is computed from the scores reported by the benchmarks. The standard deviation mentioned in this issue refers to the `stddev` reported by the JMH as shown in https://github.com/enso-org/enso/pull/10224/files#diff-e22049d31851055e89683515d5832b4804132ac981824975c8b71a0b59a3506fR43.

I believe we can possibly close this issue once:
1) #10224 is integrated.
2) The [Benchmark web app](https://enso-org.github.io/engine-benchmark-results/) displays this stddev in every graph



<blockquote><div><strong><a href="https://enso-org.github.io/engine-benchmark-results/">Enso benchmark results</a></strong></div></blockquote>
<blockquote><img src="https://avatars.githubusercontent.com/u/14013887?s=400&v=4" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/enso-org/enso/pull/10224">Gather all data from benchmarks by Akirathan · Pull Request #10224 · enso-org/enso</a></strong></div><div>Closes #9355
Pull Request Description
Gather benchmarking data in a new, JSON format. Previously, we gathered only one score value per result. Now, everything that is provided by JMH framework is g...</div></blockquote>

**Comment by JaroslavTulach** [comment]
> I believe we can possibly close this issue once:

Or we can close the issue claiming nobody really cares about benchmarks that much to need this functionality.
